scripts/common/proxy.sh

```diff
@@ -47,6 +47,55 @@ function proxy_bootstrap() {
     fi
 }
 
+# check_proxy_config tries to check if is possible connect with the registry
+# If the result is an error that is  NOT UNAUTHORIZED such as:
+# {"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":null}]}
+# Then, it means that the proxy configuration does not affected the connection and it is not invalid.
+function check_proxy_config() {
+    if [ -n "$CONTAINERD_VERSION" ]; then
+        logStep "Checking registry config with Containerd"
+
+        config_file="/etc/containerd/config.toml"
+
+        # Check if the config file exists
+        if [ ! -f "$config_file" ]; then
+            logWarn "File $config_file not found. Unable to check connection"
+            return
+        fi
+
+        # Get IP and ca_file path from the config.toml file
+        ip=$(cat "$config_file" | grep -oP '(?<=plugins\."io.containerd.grpc.v1.cri"\.registry\.configs\.").*(?="\.tls\])')
+        ca_file=$(cat "$config_file" | grep -oP '(?<=ca_file = ").*(?=")')
+
+        # Echo containerd Proxy config:
+        proxy_config_file="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+        if [ -f "$proxy_config_file" ]; then
+           echo "Containerd HTTP proxy config:"
+           grep -v -e '^\[Service\]' -e '^# Generated by kURL' "$proxy_config_file"
+           echo ""
+        fi
+
+        # Execute curl command and capture the output
+        set +e
+        echo "curl --cacert $ca_file https://$ip/v2/"
+        response=$(curl --cacert "$ca_file" "https://$ip/v2/" 2>&1)
+        curl_exit_code=$?
+        set -e
+
+        # Check if the curl command failed or response doesn't contain "UNAUTHORIZED"
+        if [ $curl_exit_code -ne 0 ] && [[ ! $response == *"UNAUTHORIZED"* ]]; then
+            logWarn "Connection with registry failed! Check the installer configuration."
+            printf "${YELLOW}\n"
+            echo "Response: ${response}"
+            printf "${NC}\n"
+            logWarn "If you are using a Proxy, ensure that it is a valid Proxy and properly configured."
+            logWarn "More info: https://kurl.sh/docs/install-with-kurl/proxy-installs"
+        else
+            logSuccess "Registry config with Containerd checked successfully"
+        fi
+    fi
+}
+
 function kubectl_no_proxy() {
```

Comment:
@laverya @emosbaugh 

In the join command I am facing an issue. If I run the code outside of kURL script it works fine but in the script it takes more time to do the curl and it is not checking the result:

Example calling the func after package_cleanup
![Screenshot 2023-05-21 at 09 32 12](https://github.com/replicatedhq/kURL/assets/7708031/23aa37f3-06f2-4833-8b27-80c3f9012e0a)


![Screenshot 2023-05-21 at 09 32 12](https://github.com/replicatedhq/kURL/assets/7708031/4756b503-47a2-4ba4-b0c7-84ae22e19a2f)

Reply:
I think it might be because +e and -e to handle the issue scenario I will try the same in another way.

Reply:
It seems sorted out the issue
Also, I change to use local vars.